funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -848,9 +848,7 @@ def cli_run():
     print("Starting HTEX Intechange")
     args = parser.parse_args()
 
-    logdir = os.path.abspath(args.logdir)
-    os.makedirs(logdir, exist_ok=True)
-    setup_logging(logfile=os.path.join(logdir, "endpoint.log"), debug=args.debug)
+    setup_logging(logfile=os.path.join(args.logdir, "endpoint.log"), debug=args.debug)
```

Comment:
is there some magic happening with this abspath? (eg is ... the interchange expecting to be in a different directory, but supplied with a relative path?) none of the other calls seem to do an abspath so is this a needed feature or some inessential fluff?

Reply:
It is inessential.

I _think_ the call was added because several lines below this, you can see another use of `abspath(args.logdir)`. I presume it was done with the intention to pass `logdir` there instead of repeating the `abspath` call.

The `setup_logging` call is going to instantiate loggers without changing with working dir, so whatever `abspath` would give in this call, that will be found when the log handlers are initialized.
Any type of FileHandler will [call abspath on its input](https://github.com/python/cpython/blob/8765b01bcf6514602affcca66e7deeeb998f9cef/Lib/logging/__init__.py#L1152).